### PR TITLE
Backports 6.0.13/s/v1

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) tickets:
- https://redmine.openinfosecfoundation.org/issues/6138
- https://redmine.openinfosecfoundation.org/issues/6139 minus the commit https://github.com/OISF/suricata/pull/8965/commits/abc76e27de7184d6f64040fb53d7d0a4d2b3df17 because it seems that ANDX support was added as a part of https://github.com/oISF/suricata/commit/2d1460622411552
which seems to have been targeted only as a feature for 7. ref: https://redmine.openinfosecfoundation.org/issues/3475#note-22

### CODE REVIEW

File: rust/src/smb/smb1_records.rs [hunk 1]
```diff
@@ -93,7 +93,7 @@ named!(pub parse_smb1_write_andx_request_record<Smb1WriteRequestRecord>,
         >>  _padding_evasion: cond!(data_offset > 36+2*(wct as u16), take!(data_offset - (36+2*(wct as u16))))
         >>  file_data: rest
         >> (Smb1WriteRequestRecord {
-                offset: if high_offset != None { ((high_offset.unwrap() as u64) << 32)|(offset as u64) } else { 0 },
+                offset: ((high_offset.unwrap_or(0) as u64) << 32) | offset as u64,
```
Comment (inashivb) [inline rust/src/smb/smb1_records.rs:96]: Please check if this change is relevant either.

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW1_files_sha256.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPR1_stats_chk | | |
| .flow.memuse | 577437888 | 712943168 | 123.47% | 

Pipeline 14493

**Comment by victorjulien** [comment]
Merged on the decode commit, smb needs a bit more thought.

**Comment by victorjulien** [comment]
In #9023

File: rust/src/smb/smb1_records.rs (hunk 1 above)
Comment (catenacyber) [reply rust/src/smb/smb1_records.rs:96]: This is relevant indeed

**Pull request closed without merging**